crates/sui-json-rpc/src/read_api.rs

```diff
@@ -147,11 +145,17 @@ impl ReadApiServer for ReadApi {
             .cloned()
             .map_or(cursor, |o_info| Some(o_info.object_id));
 
-        let data = objects.into_iter().try_fold(vec![], |mut acc, o_info| {
-            let o_resp = SuiObjectResponse::try_from((o_info, options.clone()))?;
-            acc.push(o_resp);
-            Ok::<Vec<SuiObjectResponse>, Error>(acc)
-        })?;
+        let data = match options.show_content {
```

Comment:
I will probably add an `is_not_in_object_info(&self) { self.show_bcs || self.show_content || self.show_display || self.show_storage_rebate}` method to `SuiObjectDataOptions`

https://github.com/MystenLabs/sui/blob/832ac53dfeb0636934f13d799ea4ae7187272dcc/crates/sui-json-rpc-types/src/sui_object.rs#L225-L227

```suggestion
        let data = match options.is_not_in_object_info() {
```